apps/rush-lib/src/logic/policy/PolicyValidator.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { RushConfiguration } from '../../api/RushConfiguration';
+import { GitEmailPolicy } from './GitEmailPolicy';
+import { RushPolicy } from './RushPolicy';
+
+export class PolicyValidator {
+  public static validatePolicy(rushConfiguration: RushConfiguration, bypassPolicy: boolean): void {
+    if (bypassPolicy) {
+      return;
+    }
+
+    const policies: RushPolicy[] = [new GitEmailPolicy()];
+    for (const policy of policies) {
+      policy.validate(rushConfiguration);
+    }
```

Comment:
What's the usefulness of having an array of objects, versus a hardcoded list of function calls?  code > data

Reply:
The `--bypass-policy` option might also make sense for suppressing the `nodeSupportedVersionRange` option, but even if we did that in the future, I'm not sure subclassing GitPolicy would give us much benefit. (?)

---
In reply to: [213148144](https://github.com/Microsoft/web-build-tools/pull/786#discussion_r213148144) [](ancestors = 213148144)